src/resources/documentResource.js

```diff
@@ -122,20 +122,7 @@ export function createDocumentResource(options, vm) {
           }
         },
         onSuccess(data) {
-          if (data.docs) {
-            for (let doc of data.docs) {
-              if (
-                doc.doctype === out.doctype &&
-                doc.name.toString() === out.name.toString()
-              ) {
-                out.doc = transform(doc)
-                // update data in list resources
-                updateRowInListResource(out.doctype, out.doc)
-                break
-              }
-            }
-          }
-          onSuccess?.call(vm, out.doc)
```

Comment:
why remove this? this code updates the `out.doc` with the latest values from the server.

Reply:
Makes sense. But we can update all the docs in the list.

Reply:
```javascript
data.docs
  ?.forEach((doc) => updateRowInListResource(doc.doctype, transform(doc)))
  .filter((doc) => doc.doctype === out.doctype)
  .filter((doc) => doc.name.toString() === out.name.toString())
  .forEach((doc) => out.doc = transform(doc))
```

Reply:
```javascript
data.docs
  ?.map((doc) => {
    updateRowInListResource(doc.doctype, transform(doc));
    return doc;
  })
  .filter((doc) => doc.doctype === out.doctype)
  .filter((doc) => doc.name.toString() === out.name.toString())
  .forEach((doc) => out.doc = transform(doc))
```

Reply:
Yeah, but let's keep the change to minimum in this PR. We can do other changes in a different PR.